turbopack/crates/turbopack-node/js/src/ipc/index.ts

```diff
@@ -96,12 +121,11 @@ function createIpc<TIncoming, TOutgoing>(
   });
 
   function send(message: any): Promise<void> {
-    // Reserve 4 bytes for our length prefix, we will over-write after encoding.
-    const packet = Buffer.from("0000" + JSON.stringify(message), "utf8");
-    packet.writeUInt32BE(packet.length - 4, 0);
-
     return new Promise((resolve, reject) => {
-      socket.write(packet, (err) => {
+      const stringified = JSON.stringify(message)
+      const packet = Buffer.from("0000" + stringified, "utf8");
+      packet.writeUInt32BE(packet.length - 4, 0);
+      socketWritable.write(packet, (err) => {
         process.stderr.write(`TURBOPACK_OUTPUT_D\n`);
```

Comment:
```suggestion
  function doSend(message:string): Promise<void> {
       return new Promise((resolve, reject) => {

        const packet = Buffer.from("0000" + message, "utf8");
        packet.writeUInt32BE(packet.length - 4, 0);
        socketWritable.write(packet, (err) => {
           process.stderr.write(`TURBOPACK_OUTPUT_D\n`);
           process.stdout.write(`TURBOPACK_OUTPUT_D\n`);
           if (err != null) {
            reject(err);
            } else {
            resolve();
          }
        });
    };
  }
  function send(message: any): Promise<void> {
    return doSend(JSON.stringify(message));
   }
   function sendRead(message: any): Promise<void> {
      return doSend("");
   }
```

Lets share code between send and sendReady since they are following the same protocol afaict.